bundles/org.openhab.binding.knx/src/main/java/org/openhab/binding/knx/internal/handler/IPBridgeThingHandler.java

```diff
@@ -70,20 +110,64 @@ public void initialize() {
+
+            if (!securityAvailable) {
+                logger.warn("Bridge {} missing security configuration for secure tunnel", thing.getUID());
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                        "Security configuration missing for secure tunnel");
+                return;
+            }
+            boolean tunnelOk = ((secureTunnel.user > 0) && (secureTunnel.devKey.length == 16)
+                    && (secureTunnel.userKey.length == 16));
+            if (!tunnelOk) {
+                logger.warn("Bridge {} incomplete security configuration for secure tunnel", thing.getUID());
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                        "Security configuration for secure tunnel is incomplete");
+                return;
+            }
+
+            logger.info("KNX secure tunneling needs a few seconds to establish connection");
+            // user id, key, devAuth are already stored
         } else if (MODE_ROUTER.equalsIgnoreCase(connectionTypeString)) {
             useNAT = false;
             if (ip.isEmpty()) {
                 ip = KNXBindingConstants.DEFAULT_MULTICAST_IP;
             }
-            ipConnectionType = CustomKNXNetworkLinkIP.ROUTING;
+            ipConnectionType = IPClient.IpConnectionType.ROUTER;
+        } else if (MODE_SECURE_ROUTER.equalsIgnoreCase(connectionTypeString)) {
+            useNAT = false;
+            if (ip.isEmpty()) {
+                ip = KNXBindingConstants.DEFAULT_MULTICAST_IP;
+            }
+            ipConnectionType = IPClient.IpConnectionType.SECURE_ROUTER;
+
+            if (!securityAvailable) {
+                logger.warn("Bridge {} missing security configuration for secure routing", thing.getUID());
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                        "Security configuration missing for secure routing");
+                return;
+            }
+            if (secureRouting.backboneGroupKey.length != 16) {
+                // failed to read shared backbone group key from config
+                logger.warn("Bridge {} missing security configuration for secure routing", thing.getUID());
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                        "backboneGroupKey required for secure routing; please configure");
+                return;
+            }
+            logger.info("KNX secure routing needs a few seconds to establish connection");
         } else {
-            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
-                    MessageFormat.format("Unknown IP connection type {0}. Known types are either 'TUNNEL' or 'ROUTER'",
-                            connectionTypeString));
+            logger.warn("Bridge {} unknown connection type", thing.getUID());
```

Comment:
reduce to debug

Reply:
ok, I considered setting the level to debug as hiding the error. But I am fine with reducing to debug. events.log will at complain about this as well on INFO level.